omegaconf/listconfig.py

```diff
@@ -102,17 +103,26 @@ def _validate_set(self, key: Any, value: Any) -> None:
 
         target_type = self._metadata.element_type
         value_type = OmegaConf.get_type(value)
-        if is_structured_config(target_type):
-            if (
-                target_type is not None
-                and value_type is not None
-                and not issubclass(value_type, target_type)
-            ):
-                msg = (
-                    f"Invalid type assigned : {type_str(value_type)} is not a "
-                    f"subclass of {type_str(target_type)}. value: {value}"
-                )
-                raise ValidationError(msg)
+        if target_type in (Any, None):
+            return
+
+        def raise_validation_error(
+            value_type: Any, target_type: Any, value: Any
+        ) -> None:
+            msg = (
+                f"Invalid type assigned : {type_str(value_type)} is not a "
+                f"subclass of {type_str(target_type)}. value: {value}"
+            )
+
+            raise ValidationError(msg)
+
+        if (
+            is_structured_config(value_type)
+            or is_structured_config(target_type)
+            or isinstance(value, ValueNode)
+        ):
+            if value_type is not None and not issubclass(value_type, target_type):
+                raise_validation_error(value_type, target_type, value)
```

Comment:
This is bad form:

```python
if A:
  if B:
    ...
```

you can do:

```python
if A and B:
  ...
```

Reply:
I thought in this case it would be clearer this way.